qancode.py

```diff
@@ -0,0 +1,2366 @@
+            self.task_completed = True
+            return data
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except Exception as e:
+            print('{}Exception caught: {}{}'.format(
+                bcolors.FAIL, e, bcolors.ENDC))
+        finally:
+            try:
+                self.driver.quit()
+            except AttributeError:
+                pass
+
+
+class DataManager(object):
+    def __init__(self, browsers, urls, users, task, item_types=[None], click_paths=[None], **kwargs):
+        self.browsers = browsers
+        self.urls = urls
+        self.users = users
+        self.task = task
+        self.item_types = item_types
+        self.click_paths = click_paths
+        self.all_data = []
+        self.kwargs = kwargs
+
+    def run_tasks(self):
+        for user in self.users:
+            for browser in self.browsers:
+                for url in self.urls:
+                    for item_type, click_path in zip(self.item_types, self.click_paths):
+                        retry = 5
+                        while True:
+                            if self.urls[0] == url:
+                                server_name = 'prod'
+                            else:
+                                server_name = 'RC'
+                            dw = DataWorker(browser=browser,
+                                            url=url,
+                                            user=user,
+                                            task=self.task,
+                                            item_type=item_type,
+                                            click_path=click_path,
+                                            server_name=server_name,
+                                            **self.kwargs)
+                            data = dw.run_task()
+                            if dw.task_completed:
+                                self.all_data.append({'browser': browser,
+                                                      'url': url,
+                                                      'user': user,
+                                                      'item_type': item_type,
+                                                      'click_path': click_path,
+                                                      'data': data,
+                                                      'server_name': server_name})
+                                break
+                            time.sleep(2)
+                            retry -= 1
+                            if retry < 1:
+                                print('{}WARNING: Task incomplete.{}'.format(
+                                    bcolors.FAIL, bcolors.ENDC))
+                                break
+
+
+################################################################
+# QANCODE object gets data, compares data using defined tasks. #
+################################################################
+
+
+class QANCODE(object):
```

Comment:
I would say the QANCODE object is at the heart of what makes this tool flexible and easy to use. Also it allows for cool things like integrating the QA workflow into a Jupyter notebook. Parameter file sounds ugly but maybe I'm misunderstanding.